sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventHubProducerAsyncClientTest.java

```diff
@@ -805,14 +806,15 @@ void createsEventDataBatchWithSize() {
             .assertNext(batch -> {
                 Assertions.assertNull(batch.getPartitionKey());
                 Assertions.assertTrue(batch.tryAdd(event));
+                Assertions.assertFalse(batch.tryAdd(event));
             })
             .expectComplete()
             .verify(DEFAULT_TIMEOUT);
 
         StepVerifier.create(producer.createBatch(options))
             .assertNext(batch -> {
-                Assertions.assertNull(batch.getPartitionKey());
-                Assertions.assertFalse(batch.tryAdd(tooLargeEvent));
+                assertNull(batch.getPartitionKey());
+                Assertions.assertThrows(AmqpException.class, () -> batch.tryAdd(tooLargeEvent));
```

Comment:
Throwing this exception would be breaking existing behaviour when we were assuming it would return false before.

Reply:
I agree, that fixing this is kind of outside of the scope of the original issue.
Yet javadoc for EventDataBatch.tryAdd states that:

> @throws AmqpException if {@code eventData} is larger than the maximum size of the {@link EventDataBatch}.

And this was clearly the intended behavior in the code, except that it was not working and returning false instead.
I chose to fix the issue and adjust test cases, because I dare not to change the intended behavior and javadoc, this is up to you guys.

Reply:
Hello @andris-rauda, I’ve updated this pull request to keep the validation as it used to be. 

The case that we throw AmqpException is – if the entire message cannot be accommodated in the batch, which indicate that there is no way the message can be send in the current or any new batch. Had library returned false from the tryAdd, application may assume that the tryAdd with the specific event could attempted on a new batch, it can make a loop with tryAdd infinite, so library throws since there is no way to make progress.

The Service Bus code in the main already changed to use DroppingWritableBuffer, so I updated this pull request to revert changes in that class and test changes.

Thank you so much for your contribution!